scripts/cnode-helper-scripts/cncli.sh

```diff
@@ -797,10 +805,187 @@ cncliPTsendslots() {
   done
 }
 
+#################################
+# epochdata table load process  #
+#################################
+
+getCurrNextEpoch() {
+  getNodeMetrics
+  curr_epoch=${epochnum}
+  next_epoch=$((curr_epoch+1))
+}
+
+runCurrentEpoch() {
+  getKoiosData
+  local epoch="$1"
+  echo "Processing current epoch: ${epoch}"
+  stake_param_curr="--active-stake ${active_stake_set} --pool-stake ${pool_stake_set}"
+
+  ${CNCLI} leaderlog ${cncliParams} --consensus "${consensus}" --epoch="${epoch}" ${stake_param_curr} |
+  jq -r '[.epoch, .epochNonce, .poolId, .sigma, .d, .epochSlotsIdeal, .maxPerformance, .activeStake, .totalActiveStake] | @csv' |
+  sed 's/"//g' >> "$tmpcsv"
+}
+
+runNextEpoch() {
+  getKoiosData
+  local epoch="$1"
+  echo "Processing next epoch: ${1}"
+  stake_param_next="--active-stake ${active_stake_mark} --pool-stake ${pool_stake_mark}"
+
+  ${CNCLI} leaderlog ${cncliParams} --consensus "${consensus}" --epoch="${epoch}" ${stake_param_next} |
```

Comment:
Same as other comment, either set `local epoch="$1"` at the top and use it everywhere instead of `${1}` or skip setting local variable and use `${1}`.

Reply:
I chose to widely use ${1} - (with no local epoch="$1"). It looks cleaner and less verbose and the reference to epoch number is clear by the context.

I have updated processAllEpochs as

>processAllEpochs() {
  getCurrNextEpoch
  IFS=' ' read -r -a epochs_array <<< "$EPOCHS"

  >for epoch in "${epochs_array[@]}"; do
    **set -- "$epoch"**
    if ! getConsensus "${1}"; then echo "ERROR: Failed to fetch protocol parameters for epoch ${1}."; return 1; fi
    if [[ "$1" == "$curr_epoch" ]]; then
      runCurrentEpoch ${1}
    elif [[ "$1" == "$next_epoch" ]]; then
      runNextEpoch ${1}
    else
      runPreviousEpochs ${1}
    fi
  done

Reply:
ok great, but a bit too aggressively :)
In processAllEpochs you need to use epoch as this is the variable used in the for loop. There is no input argument to this function.

Reply:
lets fix it, first time on this last change i did it as you said, then i got the ${1} fever, lol

Reply:
just fixed and pushed

>updated processAllEpochs function to use epoch var from epoch list loop, instead of input "$1"